cpu/esp_common/periph/hwrng.c

```diff
@@ -23,13 +23,40 @@
 #include "periph_conf.h"
 #include "periph/hwrng.h"
 
+#if !defined(MCU_ESP8266)
+#include "bootloader_random.h"
+#include "esp_random.h"
+#include "soc/wdev_reg.h"
+
+#define ESP32_USE_ESP_IDF_RANDOM    1
+
+#endif
+
 #define RNG_DATA_REG (*(volatile uint32_t *)RNG_DATA_REG_ADDR)
 
 void hwrng_init(void)
 {
+#if defined(MCU_ESP8266)
     /* no need for initialization */
+#else
+    if (!IS_USED(MODULE_WIFI_ANY)) {
+        /*
+         * The hardware RNG generates random numbers uses the noise in the
+         * RF system of the WiFi or the BT interface as entropy source.
+         * If both are disabled, the random number generator just returns
+         * pseudo-random numbers.
+         * However, the bootloader use an internal non-RF entropy source,
+         * the internal reference voltage noise. This can be re-enabled
+         * after startup as entropy source for applications that don't
+         * use the WiFi or the BT interface.
+         */
+        bootloader_random_enable();
+    }
+#endif
 }
 
+#if defined(MCU_ESP8266) || !defined(ESP32_USE_ESP_IDF_RANDOM)
```

Comment:
```suggestion
#if IS_USED(ESP32_USE_ESP_IDF_RANDOM)
```

Reply:
Or just split the file between esp32 and esp8266

Reply:
Ok, I can move old implementation to `cpu/esp8266` and remove it from new implementation in `cpu/esp32`. There is no reason to enable the old implementation for ESP32 because the new implementation is better because it works for all ESP32x SoC variants and gives better entropy. It is easy to split it. May I squash the split directly?